### PR TITLE
Aci doc on single container run and flags

### PR DESCRIPTION
**Pull request by gtardif**
**What I did**
* Small fixe on compose doc
* Add a separate page on single container and all `docker run` flags

**Related issue**
https://github.com/docker/compose-cli/issues/253

**(not mandatory) A picture of a cute animal, if possible in relation with what you did**

### CODE REVIEW

File: docs/aci-compose-features.md [hunk 1]
```diff
@@ -16,75 +16,81 @@ __Legend:__
...
-| service.secrets                | ✓ |  See #secrets.
-| service.security_opt           | x |  
-| service.stop_grace_period      | x |  
-| service.stop_signal            | x |  
-| service.sysctls                | x |  
-| service.ulimits                | x |  
-| service.userns_mode            | x |  
-| service.volumes                | ✓ |  Mapped to AZure File Shares. See #persistent-volumes.
+| service.pid                    | x |
+| service.ports                  | ✓ |  Only symetrical por mapping is supported in ACI. See [Exposing ports](#exposing-ports).
```
Comment (chris-crone) [inline docs/aci-compose-features.md:60]:
```suggestion
| service.ports                  | ✓ |  Only symetrical port mapping is supported in ACI. See [Exposing ports](#exposing-ports).
```

File: docs/aci-compose-features.md [hunk 2]
```diff
@@ -16,75 +16,81 @@ __Legend:__
...
+|                                |   |
+| __Config__                     | x |
+| TBD                            | x |
+|                                |   |
+
+
+## Logs
+
+Container logs can be obtained for each container with `docker logs <CONTAINER>`.
+The Docker ACI integration does not currently support aggregated logs for containers in a compose application,
```
Comment (chris-crone) [inline docs/aci-compose-features.md:88]:
```suggestion
The Docker ACI integration does not currently support aggregated logs for containers in a Compose application,
```

File: docs/aci-compose-features.md [hunk 3]
```diff
@@ -16,75 +16,81 @@ __Legend:__
...
+due to ACI not providing either aggregated logs for a container group, or log timestamps that can be easily used to reorder logs properly for display.
```
Comment (chris-crone) [inline docs/aci-compose-features.md:89]: Not sure we need this line. We can create an issue, link it above and see if users thumbs up it

File: docs/aci-container-features.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
+# Azure Container Instances: running single containerrs
+
+Single containes can be executed on ACI with the `docker run` command.
```
Comment (chris-crone) [inline docs/aci-container-features.md:3]:
```suggestion
Single containers can be executed on ACI with the `docker run` command.
```

File: docs/aci-container-features.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+- __n:__ Not yet implemented
+- __x:__ Not applicable / no available conversion
+
+| Flag                  |Map|  Notes
+|-----------------------|---|---------------------------------|
+| --cpus                | ✓ | See [Container Resources](#container-resources)
+| -d, --detach          | ✓ | Detach from container logs when container starts. By default, the command line stays attached and follow container logs
+| --domainname          | ✓ | See [Exposing ports](#exposing-ports)
+| --e, --env            | ✓ | Sets environment variable
+| --envFile             | ✓ | Sets environment variable from and external file
```
Comment (chris-crone) [inline docs/aci-container-features.md:24]: We should make this flag `--env-file` to match the existing CLI. Should I create a ticket to track?

File: docs/aci-container-features.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+| -l, --label           | x | Unsupported in Docker ACI integration, due to limitations of ACI Tags
+| -m, --memory          | ✓ | See [Container Resources](#container-resources)
+| --name                | ✓ | provide a name for the container. Name must be unique withing the ACI resource group. a name is generated by default.
```
Comment (chris-crone) [inline docs/aci-container-features.md:27]:
```suggestion
| --name                | ✓ | Provide a name for the container. Name must be unique withing the ACI resource group. a name is generated by default.
```

File: docs/aci-container-features.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+__Legend:__
+
+- __✓:__ Implemented
+- __n:__ Not yet implemented
+- __x:__ Not applicable / no available conversion
+
+| Flag                  |Map|  Notes
+|-----------------------|---|---------------------------------|
+| --cpus                | ✓ | See [Container Resources](#container-resources)
```
Comment (chris-crone) [inline docs/aci-container-features.md:20]: Should be consistent and finish all sentences with `.`

File: docs/aci-container-features.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+| -d, --detach          | ✓ | Detach from container logs when container starts. By default, the command line stays attached and follow container logs
+| --domainname          | ✓ | See [Exposing ports](#exposing-ports)
+| --e, --env            | ✓ | Sets environment variable
+| --envFile             | ✓ | Sets environment variable from and external file
+| -l, --label           | x | Unsupported in Docker ACI integration, due to limitations of ACI Tags
+| -m, --memory          | ✓ | See [Container Resources](#container-resources)
+| --name                | ✓ | provide a name for the container. Name must be unique withing the ACI resource group. a name is generated by default.
+| -p, --publish         | ✓ | See [Exposing ports](#exposing-ports). Only symetrical por mapping is supported in ACI.
```
Comment (chris-crone) [inline docs/aci-container-features.md:28]:
```suggestion
| -p, --publish         | ✓ | See [Exposing ports](#exposing-ports). Only symetrical port mapping is supported in ACI.
```

File: docs/aci-container-features.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+| --restart             | ✓ | Restart policy, must be one of: `any`, `none`, `on-failure`.
+| --rm                  | x | Not supported as [ACI does not support auto-delete containers](https://feedback.azure.com/forums/602224-azure-container-instances/suggestions/34066633-support-auto-delete-of-aci-when-container-exits-no)
+| -v, --volume          | ✓ | See [Persistent Volumes](#persistent-volumes)
+
+## Exposing ports
+
+You can expose one or several port of a container with `docker run -p <PORT>:<PORT>`
```
Comment (chris-crone) [inline docs/aci-container-features.md:35]:
```suggestion
You can expose one or more ports of a container with `docker run -p <PORT>:<PORT>`
```

File: docs/aci-container-features.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+If exposing port(s) when running a container, the corresponding ACI container group will be exposed with a public IP allocated and the required port(s) accessible.
```
Comment (chris-crone) [inline docs/aci-container-features.md:36]:
```suggestion
If ports are exposed when running a container, the corresponding ACI container group will be exposed with a public IP allocated and the required port(s) accessible.
```

File: docs/aci-container-features.md [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+> Note: [ACI does not support port mapping](https://feedback.azure.com/forums/602224-azure-container-instances/suggestions/34082284-support-for-port-mapping), so the same port number must be specified when using `-p <PORT>:<PORT>`.
+
+When exposing ports, a container can also specify the service `--domainname` flag to set a DNS hostname. `domainname` will be used to specify the ACI DNS Label Name, and the ACI container group will be reachable at `<domainname>.<region>.azurecontainer.io`.
```
Comment (chris-crone) [inline docs/aci-container-features.md:39]: nit: Consistency
```suggestion
When exposing ports, a container can also specify the service `--domainname` flag to set a DNS hostname. `domainname` will be used to specify the ACI DNS Label Name, and the ACI container group will be reachable at `<DOMAINNANE>.<REGION>.azurecontainer.io`.
```

File: docs/aci-container-features.md [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+`domainname` must be unique globally in <region>.azurecontainer.io
```
Comment (chris-crone) [inline docs/aci-container-features.md:40]:
```suggestion
`domainname` must be unique globally in <REGION>.azurecontainer.io
```

File: docs/aci-container-features.md [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+
+## Persistent volumes
+
+Docker volumes are mapped to Azure File shares, each file share is part of an Azure Storage Account.
+Volume(s) can be specified with `docker run -v <STORAGE-ACCOUNT>/<FILESHARE>:<TARGET-PATH>`.
```
Comment (chris-crone) [inline docs/aci-container-features.md:45]:
```suggestion
One or more volumes can be specified with `docker run -v <STORAGE-ACCOUNT>/<FILESHARE>:<TARGET-PATH>`.
```

File: docs/aci-container-features.md [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+
+A run command can use the `-- volume`flag several times for different volumes, several volumes can use the same or different storage accounts. Tagret path for different volume mounts must be different and not overlap.
```
Comment (chris-crone) [inline docs/aci-container-features.md:47]:
```suggestion
A run command can use the `--volume` or `-v` flag several times for different volumes. The volumes can use the same or different storage accounts. The target paths for different volume mounts must be different and not overlap.
```

File: docs/aci-container-features.md [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+There is no support for mounting a single file, or mounting a subfolder from an Azure File Share.
+
+Credentials for storage accounts will be automatically fetched at deployment time using the Azure login to retrieve the storage account key for each storage account used.
+
+## Container Resources
+
+CPU and memory reservations can be set when running containers with `docker run --cpus 1.5 --memory 2G`.
+
+It is not possible to set resource limits different than resource reservation on single containers.
```
Comment (chris-crone) [inline docs/aci-container-features.md:56]:
```suggestion
It is not possible to set resource limits that differ from resource reservation on single containers. 
```

File: docs/aci-container-features.md [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+There is no support for mounting a single file, or mounting a subfolder from an Azure File Share.
+
+Credentials for storage accounts will be automatically fetched at deployment time using the Azure login to retrieve the storage account key for each storage account used.
+
+## Container Resources
+
+CPU and memory reservations can be set when running containers with `docker run --cpus 1.5 --memory 2G`.
+
+It is not possible to set resource limits different than resource reservation on single containers. 
+ACI allows setting resource limits for containers in a container group, but these limits must stay within the reserved resources for the entire group. In the case of a single container deployed in a container group, resource limits must be equal to resource reservation.
```
Comment (chris-crone) [inline docs/aci-container-features.md:57]:
```suggestion
ACI allows setting resource limits for containers in a container group but these limits must stay within the reserved resources for the entire group. In the case of a single container deployed in a container group, the resource limits must be equal to the resource reservation.
```

File: docs/aci-container-features.md [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+
+## Logs
+
+You can view container logs with the command:
+```
+docker logs <CONTAINER-ID>
+```
```
Comment (chris-crone) [inline docs/aci-container-features.md:64]: For consistency:

```suggestion
You can view container logs with the command `docker logs <CONTAINER-ID>`.
```

File: docs/aci-container-features.md [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+
+You can follow logs with the `--follow` (`-f`) option.
+When running a container with `docker run`, by default the command line stays attached to container logs when the container starts. Use `docker run --detach` to not follow logs once the container starts.
+> Note: following ACI logs might have display issues especialy when resizing a terminal that is following container logs. This is due to ACI providing raw log pullling but no streaming of logs. Logs are effectively pulled every 2 seconds when following logs.
```
Comment (chris-crone) [inline docs/aci-container-features.md:68]:
```suggestion
> Note: Following ACI logs may have display issues especially when resizing a terminal that is following container logs. This is due to ACI providing raw log pulling but no streaming of logs. Logs are effectively pulled every 2 seconds when following logs.
```

**Review by chris-crone — APPROVED** [review]
LGTM with minor changes

File: docs/aci-container-features.md (hunk 5 above)
Comment (gtardif) [reply docs/aci-container-features.md:24]: https://github.com/docker/compose-cli/issues/802

File: docs/aci-container-features.md (hunk 7 above)
Comment (gtardif) [reply docs/aci-container-features.md:20]: updated

File: docs/aci-container-features.md (hunk 5 above)
Comment (gtardif) [reply docs/aci-container-features.md:24]: we'll update the doc with the implementation

**Pull request merged**